### DISCUSSION TITLE
Pikaday does not pass value to bound property after upgrading Livewire from 2 to 3

### DISCUSSION THREAD

**Discussion by troccoli** [livewire/livewire#6913] [1. Help]
Hello everyone.

I have a [project](https://github.com/troccoli/tfl-auto-calendar) which uses Livewire 2 and Pikaday. 

My `app.js` (actually it's in `bootstrap.js`)
```js
import 'alpinejs'
import {Pikaday} from "pikaday-momentless";
window.Pikaday = Pikaday;
```
And I have the usual `<livewire:styles />` and `<livewire:scripts />` (I'm not sure why I didn't use the Blade directive but there you go). Finally, in the Blade template for my component I initialise the date picker using the `x-init` directive
```html
<div x-data x-init="new Pikaday({field: $refs.startDate})">
  <input name="start-date" type="text" id="start-date" x-ref="startDate" wire:model.lazy="startDate"/>
</div>
```
The above is a simplified version and I actually have two date pickers in the component.

I want to upgrade to Livewire 3 and I have followed the guide, and you can find my latest attempt [here](https://github.com/troccoli/tfl-auto-calendar/tree/issue-29).

- I have removed `alpinejs` as a dependency
- I have remove both `<livewire:styles />` and `<livewire:scripts />` as they are not needed any more
- I have changed `wire.model.lazy` to `wire.model.blur` on the properties for the dates

Now, when I select the `startDate`, Livewire does an update with a `null` value, i.e. a call to`POST http://localhost/livewire/update` returns
```json
{
  "_token": "Wdinfv6GK4w0w71LEIU4JcKfx9dbh7fXh8P9ZbH8",
  "components": [
    {
      "snapshot": {
        "data": {
          "startDate": null,
          "endDate": null,
          "position": null,
          "holidayPeriods": 0
        },
        "memo": {
          "id": "hoUlitIjlErYP1gQWok2",
          "name": "generate-events",
          "path": "generate-events",
          "method": "GET",
          "children": [],
          "errors": [],
          "locale": "en"
        },
        "checksum": "d9cbd2c712505d41bbfcb3e87c35df27f461f16103b4861d4460edf64504a916"
      },
      "updates": {},
      "calls": []
    }
  ]
}
```

Can anybody help please?

**Comment by iancunicky** [comment]
I had a simmilar problem, and managed to work by adding onchange="this.dispatchEvent(new InputEvent('input'))" to the input field . Try with wire:model="startDate"
